src/lib/render-lit-html.ts

```diff
@@ -315,17 +381,19 @@ export async function* renderTemplateResult(
             }
             skipTo(attrEndOffset);
             boundAttrsCount += 1;
+            templatePartIndex++;
           }
         }
-        if (boundAttrsCount > 0) {
-          // TODO: Can we add the leading space a different way?
-          yield ` __lit-attr="${boundAttrsCount}"`;
-        }
 
-        if (writeTag) {
+        if (writeTag || boundAttrsCount > 0) {
           yield flushTo(node.sourceCodeLocation!.startTag.endOffset);
         }
 
+        if (boundAttrsCount > 0) {
+          const templatePart = templateParts[templatePartIndex];
+          yield `<!--lit-bindings ${templatePart.index}-->`;
```

Comment:
Although there may have been multiple attribute bindings, they all have the same node index so it's ok to just grab the last one's index here right (and ack that this is just for integrity purposes)?

Reply:
Also capturing IRL discussion topic: by encoding the number of bindings contained on this node in the comment, it may be possible to defer parsing the strings into an HTMLTemplateElement until the first time that new DOM actually needs to be stamped (i.e., not during hydration) as a potential performance optimization. Also requires changing Template to not eagerly parse the strings, which may be a larger refactor.

Reply:
Adding a comment here to clarify, since we can't change the name of the property.

I think that refactor might be a breaking change, so I added it as a potential change to 2.0 in my rough list.